src/ui/transform.js

```diff
@@ -0,0 +1,26 @@
+import moment from 'moment';
+import type { Message } from '../store/types';
+import { unescape } from '../utility/misc';
+
+// TODO don't mix validation and data munging
+export const messageToDetails = (email: Message) => {
+  const headers = email.payload.headers;
+
+  const fromHeader = headers.find((h) => h.name === 'From');
+  const subjectHeader = headers.find((h) => h.name === 'Subject');
+
+  if(!fromHeader) {
+    throw 'Email has no "From" header';
+  }
+  if(!subjectHeader) {
+    throw 'Email has no "Subject" header';
+  }
+
+  return {
+    id:      email.id,
+    sender:  unescape(fromHeader.value),
+    date:    moment(parseInt(email.internalDate)),
+    subject: subjectHeader.value,
+    snippet: email.snippet
+  }
```

Comment:
I was considering using decorators, too. Very clean split between formatting your data for human consumption and building a particular layout. I should be specific, though, because decorators means something different outside the Rails community. https://github.com/drapergem/draper